### PR TITLE
docs: add autocomplete documentation

### PR DESCRIPTION
**Pull request by didoesdigital**
To deliver value iteratively, we'll add autocomplete visuals later, which will make the docs much more useful and easy to understand.

Branch preview:

- https://dev.cultureamp.design/dst/add-autocomplete-documentation/components/autocomplete/

### CODE REVIEW

File: site/docs/components/autocomplete.mdx [hunk 1]
```diff
@@ -0,0 +1,105 @@
+---
+title: Autocomplete
+navTitle: Autocomplete (not built)
+summaryParagraph: Autocomplete automatically completes typed user input with matching results from a larger data set. Autosuggest breaks beyond the input provided to suggest alternative, relevant answers.
+tags: ["Autosuggest", "Live filter", "Filters", "Search", "Top bar", "Text input"]
+needToKnow:
+- 'We separate Search Field into a separate component, because it may be used for live-filtering without showing autocompleted results.'
```
Comment (dlimiter) [inline site/docs/components/autocomplete.mdx:7]: I found this a bit hard to parse. Not clear what thing the search field has been separated from and where autocomplete fits into the picture. Maybe something like  "Search and autocomplete are separated into distinct components, because  search may be used in a way that doesn't show autocompleted results." ?

File: site/docs/components/autocomplete.mdx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+- 'Unlike a Select (single- or multi-) that lets you select options from an options menu, the Autocomplete component has other behavior beyond choosing an option as the value for a setting, such as linking to other content.'
```
Comment (dlimiter) [inline site/docs/components/autocomplete.mdx:8]: Does this refer to a _select_ component or the _dropdown_ component used with a select?

File: site/docs/components/autocomplete.mdx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+Autocomplete is forgiving, letting people make mistakes such as spelling errors and still predict the content they want.
+
+### Anatomy
+
+- [Dropdown container](/components/dropdown/): the bit that drops down from a [Search Field](/components/search-field) to show Autocomplete options.
+- **Autocomplete options or items:** the options menu that lets you choose options.
+    - Each option/item may be presented as plain text or with rich media, such as images.
+
+### Behavior
```
Comment (dlimiter) [inline site/docs/components/autocomplete.mdx:30]: Do we have any guidelines around when the autocomplete kicks in, i.e., whether it is immediate (keyup on the first character) or delayed until some minimum condition (x characters, more than one match etc.). This might be framed more as a consideration related to the search type/dataset/performance.

File: site/docs/components/autocomplete.mdx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+</WhenToUse>
+<WhenNotToUse>
+
+- Avoid Autocomplete when the user has no concept of the search space and what's available. You might instead use navigation options or surface top content.
+
+</WhenNotToUse>
+</WhenToUseAndWhenNotToUse>
+
+## See also
+
```
Comment (dlimiter) [inline site/docs/components/autocomplete.mdx:88]: All these period characters following not-a-sentence list items feel a bit weird. Do we do that everywhere?

File: site/docs/components/autocomplete.mdx (hunk 3 above)
Comment (mindycultureamp) [reply site/docs/components/autocomplete.mdx:30]: +1 to above!

File: site/docs/components/autocomplete.mdx (hunk 2 above)
Comment (didoesdigital) [reply site/docs/components/autocomplete.mdx:8]: It refers to the entire select component. What could make this more useful?

Comment (dlimiter) [reply site/docs/components/autocomplete.mdx:8]: Now that I'm re-reading it I'm not sure what I was getting at here 🤦‍♂️Let's drop it :)

File: site/docs/components/autocomplete.mdx (hunk 4 above)
Comment (didoesdigital) [reply site/docs/components/autocomplete.mdx:88]: CC @berry-thanki I'd be keen to have a clear stance on this and clean it up site wide
